### PR TITLE
feat(explorer): show party assets and withdrawals on party page

### LINKED ISSUE #5277: Block Explorer: show deposit transactions in party / pubkey view
## Story

As a user with a public key
I want transactions that deposited funds to my public key to be shown in the transaction list on the party/public key page even though the transactions don't originate from my public key
So that I can see deposit transactions to my public key

## Acceptance Criteria

- [ ] I can see transactions that deposited funds to my public key in the transaction list on the party/public key page
- [ ] I can see some visual indication that the transaction is an imposter signed by someone elsee

### PR DESCRIPTION
**Pull request by edd**
- **feat(explorer): add party deposits and withdrawals panel showing code**
- **feat(explorer): properly render as a table of the first 5**

Closes #5277 

# Demo

## Preview of the last 5
Screenshot 2024-05-29 at 17 12 35

### CODE REVIEW

**Comment by edd** [comment]
Still WIP, getting there... 
Screenshot 2024-06-14 at 17 53 21
Screenshot 2024-06-14 at 17 53 27
Screenshot 2024-06-14 at 17 53 31

File: apps/explorer/src/app/routes/parties/id/components/Party-deposits-withdrawals.graphql
Comment (mattrussell36) [inline apps/explorer/src/app/routes/parties/id/components/Party-deposits-withdrawals.graphql:1]: .graphql file convention is PartyDepositsWithdrawals.graphql

File: apps/explorer/src/app/routes/parties/id/components/party-accounts.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const PartyAccounts = ({ partyId }: PartyAccountsProps) => {
   const accounts =
     party?.accountsConnection?.edges?.filter((edge) => edge?.node) || [];
   return (
-    <div className="block min-h-44 h-60 4 w-full border-red-800 relative">
+    <div className="block min-h-44 4 w-full border-red-800 relative">
```
Comment (mattrussell36) [inline apps/explorer/src/app/routes/parties/id/components/party-accounts.tsx:26]: Not sure if you actually want it but we have vega-red

File: apps/explorer/src/app/routes/parties/id/components/lib/combine-deposits-withdrawals.test.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { getDepositWithdrawalStatusLabel } from '../party-deposits-withdrawals-status-icon';
+import { isDepositStatus } from './combine-deposits-withdrawals';
+import { DepositStatus, WithdrawalStatus } from '@vegaprotocol/types';
+
+describe('isDepositStatus & getDepositWithdrawalStatusLabel', () => {
+  it('Use deposit status label for open, regardless of type', () => {
+    expect(isDepositStatus(DepositStatus['STATUS_OPEN'])).toBe(true);
```
Comment (mattrussell36) [inline apps/explorer/src/app/routes/parties/id/components/lib/combine-deposits-withdrawals.test.ts:7]: non blocking: I find these to be overkill given we are using typescript and we inherently test this via getDepositWithdrawalStatusLabel

File: apps/explorer/src/app/routes/parties/id/components/party-block-deposits.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  const navigate = useNavigate();
+
+  const { data, error, loading } = useExplorerPartyDepositsWithdrawalsQuery({
+    variables: { partyId },
+  });
+
+  const sortedData = data ? combineDepositsWithdrawals(data) : [];
+
+  const { screenSize } = useScreenDimensions();
+  const isTruncated = useMemo(
```
Comment (mattrussell36) [inline apps/explorer/src/app/routes/parties/id/components/party-block-deposits.tsx:52]: Only use useMemo if you actually having something intensive enough. Usually this is worse for perf

Theres a few other spots in this PR where I'd remove useMemo too

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/explorer/src/app/routes/parties/id/components/party-block-deposits.tsx (hunk 3 above)
Comment (edd) [reply apps/explorer/src/app/routes/parties/id/components/party-block-deposits.tsx:52]: Removing them 👍

File: apps/explorer/src/app/routes/parties/id/components/lib/combine-deposits-withdrawals.test.ts (hunk 2 above)
Comment (edd) [reply apps/explorer/src/app/routes/parties/id/components/lib/combine-deposits-withdrawals.test.ts:7]: I can see that point of view, but these are testing (and demonstrating via tests) the situation that leads to the label. Even with the type narrowing the shared labels will come out as a `DepositStatus`, and that's expected and handled correctly (even if it seems surprising). I'm going to leave these as is.

File: apps/explorer/src/app/routes/parties/id/components/party-accounts.tsx (hunk 1 above)
Comment (edd) [reply apps/explorer/src/app/routes/parties/id/components/party-accounts.tsx:26]: a) I didn't realise, thanks for pointing that out
b) I'm not even sure why this class is here, so I have removed the red border entirely.

File: apps/explorer/src/app/routes/parties/id/components/Party-deposits-withdrawals.graphql
Comment (edd) [reply apps/explorer/src/app/routes/parties/id/components/Party-deposits-withdrawals.graphql:1]: Fixing

**Review by mattrussell36 — APPROVED** [review]

**Review by bwallacee — APPROVED** [review]

**Review by MadalinaRaicu — APPROVED** [review]

**Pull request merged**